CCL.Importer/Processing/GrabberProcessor.cs

```diff
@@ -0,0 +1,137 @@
+﻿using CCL.Types.Components;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.Linq;
+using System.Reflection;
+using UnityEngine;
+
+namespace CCL.Importer.Processing
+{
+    [Export(typeof(IModelProcessorStep))]
+    internal class GrabberProcessor : ModelProcessorStep
+    {
+        private class CachedResource<T> where T : UnityEngine.Object
+        {
+            private Dictionary<string, T>? _cachedResources = null;
+
+            public Dictionary<string, T> Cache
+            {
+                get
+                {
+                    if (_cachedResources == null)
+                    {
+                        BuildCache();
+                    }
+
+                    return _cachedResources!;
+                }
+            }
+
+            private int _expectedResources;
+
+            public CachedResource(int expectedResources)
+            {
+                _expectedResources = expectedResources;
+            }
+
+            public void BuildCache()
+            {
+                _cachedResources = Resources.FindObjectsOfTypeAll<T>().ToDictionary(k => k.name, v => v);
+
+                CCLPlugin.Log($"{typeof(T).Name} cache created with {_cachedResources.Count} sounds.");
+
+                if (_cachedResources.Count < SoundGrabber.SoundNames.Length)
+                {
+                    CCLPlugin.Error($"Possible miscreation of {typeof(T).Name} cache ({_cachedResources.Count}/{_expectedResources})");
+                }
+            }
+
+            public void ClearCache()
+            {
+                if (_cachedResources != null)
+                {
+                    _cachedResources.Clear();
+                    _cachedResources = null;
+                }
+            }
+        }
+
+        private static CachedResource<AudioClip> s_soundCache = new CachedResource<AudioClip>(SoundGrabber.SoundNames.Length);
+
+        public override void ExecuteStep(ModelProcessor context)
+        {
+            ProcessSoundsOnPrefab(context.Car.prefab);
```

Comment:
Good location to use the new prefab iterator on context.Car

Reply:
Yeah, but it'll have to be after the merge since I don't have it available yet. I'll also get rid of `ProcessSoundsOnPrefab` and just call it directly since it'll be a loop and not 5 different calls anymore.